src/App.tsx

```diff
@@ -8,53 +8,116 @@
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useState } from 'react';
+import { User } from './types/User';
+import { getPosts, getUsers } from './api/api';
+import { Post } from './types/Post';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [showSideBar, setShowSideBar] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => setIsError(true));
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    if (activeUser) {
+      setLoading(true);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+      getPosts(activeUser)
+        .then(setPosts)
+        .catch(() => setIsError(true))
```

Comment:
Similar to the previous comment, the error handling for the `getPosts` call is minimal. It would be beneficial to provide more detailed error messages or logging.